javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/collector/ReferenceCollectingClassVisitor.java

```diff
@@ -276,6 +281,10 @@ private static ManifestationFlag computeTypeManifestationFlag(int access) {
   private class AdviceReferenceMethodVisitor extends MethodVisitor {
     private int currentLineNumber = -1;
 
+    // this data structure will remember last two LDC <class> instructions before
+    // InstrumentationContext.get() call
+    private final FixedSizeQueue<String> lastTwoClassConstants = new FixedSizeQueue<>(2);
+
```

Comment:
it might work nice to extract this to a separate "TrackingMethodVisitor", and then `new AdviceReferenceMethodVisitor(new TrackingMethodVisitor(methodVisitor))`

Reply:
That's a great idea!